src/transactions/TransactionFrame.cpp

```diff
@@ -1660,6 +1661,24 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
                 subSeedSha.add(sorobanBasePrngSeed);
                 subSeedSha.add(xdr::xdr_to_opaque(opNum));
                 subSeed = subSeedSha.finish();
+
+                // emit/accumulate soroban metrics, now that the tx has been
+                // validated and the op will get applied
+                SorobanMetrics& metrics =
```

Comment:
Even though there is only a single Soroban operation per transaction, it would still be cleaner if we define this outside of the loop. This probably belongs to `apply` function and it could be called before calling `applyOperations`.

Reply:
Moved the metrics outside of the ops apply loop, before the `applyOperations` call.